demomultistoreform/demomultistoreform.php

```diff
@@ -0,0 +1,131 @@
+        $this->need_instance = 0;
+
+        $this->bootstrap = true;
+        parent::__construct();
+
+        $this->displayName = 'Demo multistore configuration form';
+        $this->description = 'Module created for the purpose of showing an example of multistore configuration form';
+        $this->ps_versions_compliancy = array('min' => '1.7.8.0', 'max' => _PS_VERSION_);
+    }
+
+    /**
+     * @return bool
+     */
+    public function install(): bool
+    {
+        return $this->getInstaller()->createTables()
+            && parent::install()
+            && $this->registerHook('displayFooterBefore');
+    }
+
+    /**
+     * @return bool
+     */
+    public function uninstall(): bool
+    {
+        return $this->getInstaller()->dropTables() && parent::uninstall();
+    }
+
+    public function getContent(): void
+    {
+        $route = SymfonyContainer::getInstance()->get('router')->generate('demo_multistore');
+        Tools::redirectAdmin($route);
+    }
+
+    /**
+     * Gets the ContentBlockInstaller from service container if possible (at uninstall),
+     * otherwise instantiate class directly (at install)
+     *
+     * @return ContentBlockInstaller
+     */
+    private function getInstaller(): ContentBlockInstaller
+    {
+        try {
+            $installer = $this->get('prestashop.module.demo_multistore.content_block_installer');
+        } catch (Exception $e) {
+            $installer = null;
+        }
+
+        if (empty($installer)) {
+            $installer = new ContentBlockInstaller(
+                $this->get('doctrine.dbal.default_connection'),
+                $this->getContainer()->getParameter('database_prefix')
+            );
+        }
+        return $installer;
+    }
+
+    /**
+     * @param array $params
+     *
+     * @return string
+     */
+    public function hookdisplayFooterBefore($params): string
+    {
+        $this->context->smarty->assign([
+            'contentBlockList' => $this->getContentBlocksForFrontOffice(),
+            'color' => Configuration::get('PS_DEMO_MULTISTORE_COLOR'),
+            'italic' => Configuration::get('PS_DEMO_MULTISTORE_ITALIC'),
+            'bold' => Configuration::get('PS_DEMO_MULTISTORE_BOLD'),
+        ]);
+
+        return $this->display(__FILE__, 'views/templates/hook/displayBlockContent.tpl');
+    }
+
+    /**
+     * @return array
+     */
+    private function getContentBlocksForFrontOffice(): array
+    {
+        $dbPrefix = $this->getContainer()->getParameter('database_prefix');
+        $connexion = $this->get('doctrine.dbal.default_connection');
+        $qb = $connexion->createQueryBuilder()
+            ->from($dbPrefix . 'content_block', 'cb')
+            ->select('cb.id_content_block, cb.title, cb.description, cb.enable')
+            ->join('cb', $dbPrefix . 'content_block_shop', 'cbs', 'cbs.id_content_block = cb.id_content_block')
+            ->where('cbs.id_shop = ' . $this->context->shop->id  . ' and cb.enable = true');
```

Comment:
```suggestion
            ->where('cbs.id_shop = ' . (int) $this->context->shop->id  . ' and cb.enable = true');
```

good practices ;)

Reply:
better one
```
->where('cbs.id_shop = :shopId AND cb.enable = true')
->setParameter('shopId', (int) $this->context->shop->id)
```
setParameter also accepts $type as third argument, but I don't think its necessary here. (As I see we only use it if its INT_ARRAY or so)